iroh-bytes/src/store/fs.rs

```diff
@@ -1611,65 +1611,93 @@ impl ActorState {
             EntryState::Complete {
                 data_location,
                 outboard_location,
-            } => {
-                match data_location {
-                    DataLocation::Inline(()) => {
-                        // ignore export mode, just copy. For inline data we can not reference anyway.
-                        let data = tables.inline_data.get(temp_tag.hash())?.ok_or_else(|| {
-                            ActorError::Inconsistent("inline data not found".to_owned())
-                        })?;
-                        tracing::trace!("exporting inline data to {}", target.display());
-                        tx.send(std::fs::write(&target, data.value()).map_err(|e| e.into()))
-                            .ok();
-                    }
-                    DataLocation::Owned(size) => {
-                        let path = self.options.path.owned_data_path(temp_tag.hash());
-                        if mode == ExportMode::Copy {
+            } => match data_location {
+                DataLocation::Inline(()) => {
+                    // ignore export mode, just copy. For inline data we can not reference anyway.
+                    let data = tables.inline_data.get(temp_tag.hash())?.ok_or_else(|| {
+                        ActorError::Inconsistent("inline data not found".to_owned())
+                    })?;
+                    tracing::trace!("exporting inline data to {}", target.display());
+                    tx.send(std::fs::write(&target, data.value()).map_err(|e| e.into()))
+                        .ok();
+                }
+                DataLocation::Owned(size) => {
+                    let path = self.options.path.owned_data_path(temp_tag.hash());
+                    match mode {
+                        ExportMode::Copy => {
                             // copy in an external thread
                             self.rt.spawn_blocking(move || {
                                 tx.send(export_file_copy(temp_tag, path, size, target, progress))
                                     .ok();
                             });
-                        } else {
-                            match std::fs::rename(&path, &target) {
-                                Ok(()) => {
-                                    let entry = EntryState::Complete {
-                                        data_location: DataLocation::External(vec![target], size),
-                                        outboard_location,
-                                    };
-                                    drop(guard);
-                                    tables.blobs.insert(temp_tag.hash(), entry)?;
-                                    drop(temp_tag);
-                                    tx.send(Ok(())).ok();
-                                }
-                                Err(e) => {
+                        }
```

Comment:
Not sure, maybe this whole procedure is long enough to warrant a fn?

Reply:
I tried, and ran into borrowchk issues and then aborted 😓